cg/services/post_processing/abstract_classes.py

```diff
@@ -0,0 +1,91 @@
+"""Post-processing service abstract classes."""
+
+from abc import abstractmethod, ABC
+from pathlib import Path
+
+from pydantic import BaseModel
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.services.post_processing.abstract_models import PostProcessingDTOs, RunMetrics, RunData
+from cg.store.store import Store
+
+
+class RunDataGenerator(ABC):
+    """Abstract class for that holds functionality to create a run data model."""
+
+    @abstractmethod
+    def _validate_run_name(self, run_name: str) -> None:
```

Comment:
Generally an underscore denotes that a method is private and is only to be used internally. If it is a method that will only be used internally, it does not make sense to make it part of the design contract

Reply:
Yes, that is a good point, maybe this function does not need to be in the parent class

Reply:
fixed